go/tools/builders/nogo_change.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"go/token"
+	"os"
+	"path/filepath"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+// Edit describes the replacement of a portion of a text file.
+type Edit struct {
+	New   string `json:"new"`   // the replacement
+	Start int    `json:"start"` // starting byte offset of the region to replace
+	End   int    `json:"end"`   // ending byte offset of the region to replace
+}
+
+// AnalyzerEdits represents the mapping of files to their edits for a specific analyzer.
+type AnalyzerEdits struct {
+	FileToEdits map[string][]Edit `json:"file_to_edits"` // File path as the key, edits as the value
+}
+
+// Change represents a collection of analyzer edits.
+type Change struct {
+	AnalyzerToEdits map[string]AnalyzerEdits `json:"analyzer_to_edits"` // Analyzer as the key, files-to-edits mapping as the value
+}
+
+// DiagnosticEntry represents a diagnostic entry with the corresponding analyzer.
+type DiagnosticEntry struct {
+	analysis.Diagnostic
```

Comment:
Oh, I see, you just moved an existing struct here. That's fine then, but you should still make this struct private, otherwise people may start using it and we will run into issues mentioned in the link above

Reply:
done